com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs

```diff
@@ -1251,6 +1251,32 @@ void SetupVignette(Material material)
                 m_Vignette.smoothness.value * 5f
             );
 
+#if ENABLE_VR && ENABLE_XR_MODULE
+            if (xrPass != null && xrPass.enabled)
+            {
+                var v3 = new Vector4(0.0f, 0.0f, 0.0f, 0.0f);
+                float centerDeltaX = 0.5f - center.x;
+                float centerDeltaY = 0.5f - center.y;
+
+                XRView view0 = xrPass.views[0];
+                v2.x = view0.eyeCenterUV.x - centerDeltaX;
+                v2.y = view0.eyeCenterUV.y - centerDeltaY;
+                if (xrPass.views.Count == 1)
+                {
+                    v3.z = v2.x;
+                    v3.w = v2.y;
+                }
+                else
+                {
+                    XRView view1 = xrPass.views[1];
+                    v3.x = view1.eyeCenterUV.x - centerDeltaX;
+                    v3.y = view1.eyeCenterUV.y - centerDeltaY;
+                }
+
+                material.SetVector(ShaderConstants._Vignette_Params3, v3);
```

Comment:
Even this part should really only be set if we're in multiview/spi situation (viewCount > 1)

Reply:
Yes, good point, I'll make that part of the code only used in single-pass mode and run some tests.